source/intro.tex

```diff
@@ -919,23 +919,42 @@
 Certain other operations are described in this document as
 undefined behavior (for example, the effect of
 attempting to modify a const object).
+
+\pnum
+\indextext{observable checkpoint}
+Certain events in the execution of a program
+are termed \defn{observable checkpoints}.
```

Comment:
... and use \defnadj{observable}{checkpoint}

You get the indexing for free, then.